qiskit/aqua/operators/primitive_ops/pauli_op.py

```diff
@@ -70,10 +71,21 @@ def add(self, other: OperatorBase) -> OperatorBase:
         if isinstance(other, PauliOp) and self.primitive == other.primitive:
             return PauliOp(self.primitive, coeff=self.coeff + other.coeff)
 
+        if (
+                isinstance(other, PauliOp)
+                and isinstance(self.coeff, (int, float, complex))
+                and hasattr(other, 'coeff')
+                and isinstance(other.coeff, (int, float, complex))
+        ):
+            return self.to_summed_pauli_op() + other.to_summed_pauli_op()
```

Comment:
I guess here the SparsePauliOp could be directly created with the two paulis in it to save the conversions and copies. Maybe the efficiency here is not so much an issue